### PR TITLE
feat(swap): fetch swap proposals

### LINKED ISSUE #20340: Trigger get swap proposal on parameter change
All the flows that call the getSuggestedRoutes endpoint based on Swap parameters changes
Trigger the appropriate loading states
Store the response from the SuggestedRoutes somewhere (processing that and showing it to the user handled in a separate task)

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #20340

### Summary

This PR integrates fetching swap proposals from status-go endpoints into the setup swap screen

https://github.com/user-attachments/assets/7a6ec5f3-b74f-4fe7-be64-2981f75f66e7

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

### Steps to test

- Open Status
- Log in with a user with some funds
- Go to Wallet
- Go to account
- Tap on Swap button
- Select asset to pay
- Enter an amount and wait for a swap proposal to be fetched

status: ready

### CODE REVIEW

File: src/quo/components/banners/alert_banner/view.cljs
Comment (J-Son89) [inline src/quo/components/banners/alert_banner/view.cljs]: why not just pass a map of container-style like all other quo components?
This creates a consistent api and devs kind of know what to expect then.

File: src/status_im/contexts/wallet/swap/setup_swap/view.cljs
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/swap/setup_swap/view.cljs]: Constant namespace is a better home for these variables.

File: src/quo/components/banners/alert_banner/view.cljs
Comment (briansztamfater) [reply src/quo/components/banners/alert_banner/view.cljs]: Makes sense, just updated the implementation!

File: src/status_im/contexts/wallet/swap/setup_swap/view.cljs
Comment (briansztamfater) [reply src/status_im/contexts/wallet/swap/setup_swap/view.cljs]: Moved

**Review by shivekkhurana — APPROVED** [review]

File: src/status_im/contexts/wallet/swap/events.cljs
Comment (clauxx) [inline src/status_im/contexts/wallet/swap/events.cljs:75]: Are they intended to be the same?

File: src/utils/number.cljs
Comment (smohamedjavid) [inline src/utils/number.cljs:27]: Should we add a nil check? to prevent any unwanted crash or behaviour.

File: src/utils/money.cljs
Comment (smohamedjavid) [inline src/utils/money.cljs:223]: Just curious: any reason why we use `2`? I have a feeling this could change depending on the currency decimal because some currencies will have higher decimals.

Comment (briansztamfater) [reply src/utils/money.cljs:223]: Just because it is conversion to fiat, we don't need more than 2 decimals

File: src/utils/number.cljs
Comment (briansztamfater) [reply src/utils/number.cljs:27]: We can add it for sure, why not :)

File: src/status_im/contexts/wallet/swap/events.cljs
Comment (briansztamfater) [reply src/status_im/contexts/wallet/swap/events.cljs:75]: Do you mean `to-address` and `from-address`? Yes, they are intended to be the same. We can remove those bindings and add use `wallet-address` directly, or keep them in separate variables to improve readability (no strong opinions on thisn tbh).

**Review by alwx — APPROVED** [review]

File: src/status_im/contexts/wallet/swap/events.cljs
Comment (clauxx) [inline src/status_im/contexts/wallet/swap/events.cljs:77]: are these expected to diverge at any point? If not, would be better to have one binding instead

File: src/status_im/contexts/wallet/swap/setup_swap/view.cljs
Comment (clauxx) [inline src/status_im/contexts/wallet/swap/setup_swap/view.cljs:172]: can use `->`

**Review by clauxx — APPROVED** [review]

**Comment by pavloburykh** [comment]
@briansztamfater thanks for the PR. I am facing same [issue](https://github.com/status-im/status-mobile/pull/21134#issuecomment-2318412824) in current PR. Unable to proceed testing until the fix is merged.

**Comment by briansztamfater** [comment]
@pavloburykh rebased to develop, that should solve the issue as it includes the status-go commit with the fix

**Comment by pavloburykh** [comment]
> @pavloburykh rebased to develop, that should solve the issue as it includes the status-go commit with the fix

@briansztamfater thanks. I guess there is some miscommunication here: I was referring to the [ISSUE 2](https://github.com/status-im/status-mobile/pull/21134#issuecomment-2318412824) reported by Volodymyr about the broken keyboard. 

Issue is reproducible in this PR as well, rebasing didn't fix it. 

From what I see now Android is affected. 

**Android:** 

https://github.com/user-attachments/assets/397b303e-671f-40d4-8aa9-bf1be5864038

**Comment by pavloburykh** [comment]
@briansztamfater I was able to test on IOS. Here are some questions:


### Question 1 

Is it expected for now that SNT is the only token I can swap TO? I see that token switching has not been implemented yet but want to make sure I am not missing something


https://github.com/user-attachments/assets/60309248-e64e-414a-a900-2c806acefd0c

**Comment by pavloburykh** [comment]
### Question 2 

When trying to swap tokens which have low balance I constantly get `Something is wrong` error. Could you please check the logs and see which is the actual error (I believe most likely there is not enough tokens to cover fee). And the second question: is proper error handling should be in scope of this PR or there is a separate ticket which we will implement in future?  

[logs (45).zip](https://github.com/user-attachments/files/16846525/logs.45.zip)

https://github.com/user-attachments/assets/54d0fa14-1f67-43a8-b044-06b74ebaa6ff

**Comment by briansztamfater** [comment]
@pavloburykh I see, sorry for the misunderstood. Probably as #21134 is being tested by Volodymyr maybe it is better to test it there? I mean, if trigger swaps works, it means swap proposals also works. I can fix issues here too, but maybe it is better to just test one PR so we don't overlap QA efforts. wdyt?
In that PR I've set DAI as the default receive token (until we implement receive token selector) and fixed the Android bug with the keyboard.

**Comment by pavloburykh** [comment]
>  I can fix issues here too, but maybe it is better to just test one PR so we don't overlap QA efforts. wdyt?

@briansztamfater totally agree. So can we close this PR as same changes will be merged in scope of https://github.com/status-im/status-mobile/pull/21134 ?

**Comment by briansztamfater** [comment]
> > I can fix issues here too, but maybe it is better to just test one PR so we don't overlap QA efforts. wdyt?
> 
> @briansztamfater totally agree. So can we close this PR as same changes will be merged in scope of #21134 ?

I'd prefer to merge this and #21076 so linked issues are updated too. And waiting for green light on #21134 which will ensure issues are tested.

**Comment by briansztamfater** [comment]
@pavloburykh updated the labels to reflect what we agreed above. Let me know if we can proceed with merge!

**Comment by pavloburykh** [comment]
Thank you @briansztamfater! E2E failure is not PR related. Ready for merge.

**Pull request merged**
